### PR TITLE
Add RW61x monolithic build support on BLE applications

### PR DESCRIPTION
**Pull request by yeaissa**
The feature allows to link the binary blobs with BLE applications during the build, forming
one single monolithic image that can be flashed directly to the board.

### CODE REVIEW

File: tests/bluetooth/tester/testcase.yaml [hunk 1]
```diff
@@ -6,9 +6,16 @@ tests:
       - native_posix
       - native_sim
       - nrf52840dk/nrf52840
-      - rd_rw612_bga
     tags: bluetooth
     harness: bluetooth
+  bluetooth.general.tester.nxp:
+    # Disabling monolithic since CI environment doesn't use blobs
```
Comment (dleach02) [inline tests/bluetooth/tester/testcase.yaml:12]: is it the "build_only" setting that does this?

Comment (dleach02) [reply tests/bluetooth/tester/testcase.yaml:12]: nevermind... I see the disabling of the monolithic kconfig setting

**Comment by decsny** [comment]
why do you have the wording as linked "dynamically", what is dynamic about it? "dynamic linking" has a [specific meaning](https://en.wikipedia.org/wiki/Dynamic_linker), but I don't think this is doing dynamic linking, am I misunderstanding something?

**Comment by yeaissa** [comment]
@decsny, this feature allows to link the binary blob with the application which forms a single monolithic image that can be flashed directly to the board. Without it, the user has to flash the binary blob manually since it won't be linked to the application image.
We have added a doc section for BLE on the latest commit, please give it a look.

**Comment by decsny** [comment]
> @decsny, this feature allows to link the binary blob with the application which forms a single monolithic image that can be flashed directly to the board. Without it, the user has to flash the binary blob manually since it won't be linked to the application image.
> We have added a doc section for BLE on the latest commit, please give it a look. 

I understand what the PR is doing, what I'm saying is in your commit messages you are saying "dynamic linking" which I don't think is what you are really doing

**Comment by yeaissa** [comment]
> > @decsny, this feature allows to link the binary blob with the application which forms a single monolithic image that can be flashed directly to the board. Without it, the user has to flash the binary blob manually since it won't be linked to the application image.
> > We have added a doc section for BLE on the latest commit, please give it a look.
> 
> I understand what the PR is doing, what I'm saying is in your commit messages you are saying "dynamic linking" which I don't think is what you are really doing

@decsny you are right. Updated the commit messages, and removed the misleading "dynamic linking".

**Review by dleach02 — DISMISSED** [review]

File: boards/nxp/rd_rw612_bga/doc/index.rst [hunk 2]
```diff
@@ -134,6 +146,26 @@ should see the following message in the terminal:
    ***** Booting Zephyr OS zephyr-v3.6.0 *****
    Hello World! rd_rw612_bga
 
+Bluetooth
+=========
+
+BLE functionality requires to fetch binary blobs, so make sure to follow
+the ``Fetch Binary Blobs`` section first.
+
+Those binary blobs can be used in two different ways, depending if ``CONFIG_NXP_MONOLITHIC_BT``
```
Comment (kartben) [inline boards/nxp/rd_rw612_bga/doc/index.rst:155]:
```suggestion
Those binary blobs can be used in two different ways, depending if :kconfig:option:`CONFIG_NXP_MONOLITHIC_BT`
```

File: boards/nxp/rd_rw612_bga/doc/index.rst [hunk 3]
```diff
@@ -134,6 +146,26 @@ should see the following message in the terminal:
...
+is enabled or not:
+
+- ``CONFIG_NXP_MONOLITHIC_BT`` is enabled (default):
```
Comment (kartben) [inline boards/nxp/rd_rw612_bga/doc/index.rst:158]:
```suggestion
- :kconfig:option:`CONFIG_NXP_MONOLITHIC_BT` is enabled (default):
```

File: boards/nxp/rd_rw612_bga/doc/index.rst [hunk 4]
```diff
@@ -134,6 +146,26 @@ should see the following message in the terminal:
...
+
+The required binary blob will be linked with the application image directly, forming
+one single monolithic image.
+The user has nothing else to do other than flashing the application to the board.
+
+- ``CONFIG_NXP_MONOLITHIC_BT`` is disabled:
```
Comment (kartben) [inline boards/nxp/rd_rw612_bga/doc/index.rst:164]:
```suggestion
- :kconfig:option:`CONFIG_NXP_MONOLITHIC_BT` is disabled:
```

**Comment by yeaissa** [comment]
Updated, Thanks @kartben for the suggestions.

**Review by dleach02 — DISMISSED** [review]

**Comment by yeaissa** [comment]
@jhedberg, Could you please review this PR ?

**Review by jhedberg — DISMISSED** [review]

**Review by danieldegrasse — DISMISSED** [review]

**Comment by yeaissa** [comment]
Updated west sha.

**Comment by dleach02** [comment]
@jhedberg we merged the HAL change which purged the approvals...

**Review by dleach02 — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**